src/SimpleSoft.Database.EFCore/EFCoreContextContainer.cs

```diff
@@ -102,6 +103,58 @@ CancellationToken ct
             if (_options.AutoSaveChanges)
                 await SaveChangesAsync(ct).ConfigureAwait(false);
 
+            if (!_options.NoTracking)
+                return result;
+
+            var entry = _context.Entry(result);
+            if (entry.State != EntityState.Detached)
```

Comment:
To make the code simpler, I don't think there's a need to check for the state, just set it as `Detached`